shop/src/components/payment/crypto/CryptoDirect.js

```diff
@@ -92,22 +95,44 @@ const PayWithCryptoDirect = ({ submit, encryptedData, onChange, loading }) => {
 
   const acceptedTokens = config.acceptedTokens
 
+  useEffect(() => {
+    async function queryWalletBalance() {
+      return await wallet.signer.getBalance()
+    } //returns a promise that, if resolved, will give the balance of the wallet in wei (BigNumber)
+
+    onChange(
+      queryWalletBalance()
```

Comment:
Why is it in a separate function, can't we just use `wallet.signer.getBalance().then(...)`?